utils/sparse.py

```diff
@@ -158,3 +162,56 @@ def _apply_one_shot(self):
             _LOGGER.info(f"Training recipe for one-shot application not recognized by the manager. Got recipe: "
                          f"{self.train_recipe}"
                          )
+
+    def save_sample_inputs_outputs(
+        self,
+        dataloader: "Dataloader",  # flake8 : noqa F8421
+        num_export_samples=100,
+        save_dir: Optional[str] = None,
+    ):
+
+        save_dir = save_dir or ""
+        if not dataloader:
+            raise ValueError(
+                f"Expected a data loader for exporting samples. Got {dataloader}"
+            )
+
+        device = "cuda" if torch.cuda.is_available() else "cpu"
+        exported_samples = 0
+
+        sample_in_dir = os.path.join(save_dir, "sample-inputs")
+        sample_out_dir = os.path.join(save_dir, "sample-outputs")
+
+        os.makedirs(sample_in_dir, exist_ok=True)
+        os.makedirs(sample_out_dir, exist_ok=True)
+
+        for _, (images, _, _, _) in enumerate(dataloader):
+            images = (
+                images.float() / 255
+            )  # uint8 to float32, 0-255 to 0.0-1.0
+
+            device_imgs = images.to(device, non_blocking=True)
+            outs = self.model(device_imgs)
+
+            # Move to cpu for exporting
+            ins = images.detach().to("cpu")
+            outs = [elem.detach().to("cpu") for elem in outs]
+            outs_gen = zip(*outs)
+
+            for sample_in, sample_out in zip(ins, outs_gen):
+                sample_out = list(sample_out)
+                file_idx = f"{exported_samples}".zfill(4)
+
+                numpy.savez(f"{sample_in_dir}/inp-{file_idx}.npz", sample_in)
+                numpy.savez(f"{sample_out_dir}/out-{file_idx}.npz", *sample_out)
```

Comment:
are these being saved as dictionaries?

Reply:
No, YOLO returns a list of 3 items as output for each image, that's why the `*`

Reply:
Good point. Default behavior on ONNX export is to name the outputs `output_0`...`output_{n}`.  Let's make it a dict with those names to match ONNX

Reply:
I checked the Sample Outputs from SparseZoo and they follow the same default numpy names